pkg/scheduler/framework/plugins/imagelocality/image_locality.go

```diff
@@ -63,7 +64,11 @@ func (pl *ImageLocality) Score(ctx context.Context, state *framework.CycleState,
 	}
 	totalNumNodes := len(nodeInfos)
 
-	score := calculatePriority(sumImageScores(nodeInfo, pod.Spec.Containers, totalNumNodes), len(pod.Spec.Containers))
+	scoredImages := sets.New[string]()
+	imageScores := sumImageScores(nodeInfo, pod.Spec.InitContainers, &scoredImages, totalNumNodes)
```

Comment:
@denkensk I optimized it, please review it.

Reply:
@xiaomudk  I think it's better to refactor the func `sumImageScores`（it's just used in image_locality.go） as 
```go
func sumImageScores(nodeInfo *framework.NodeInfo, initContainers []v1.Container, containers []v1.Container, totalNumNodes int) int64 {

}
```

Reply:
+1 or pass in the pod directly.

Reply:
@denkensk @kerthcet  Thinks. done